src/mono/mono/metadata/mono-debug.c

```diff
@@ -1067,88 +1068,73 @@ mono_is_debugger_attached (void)
-}
-
-static gboolean
-bsymfile_match (BundledSymfile *bsymfile, const char *assembly_name)
-{
-	if (!strcmp (bsymfile->aname, assembly_name))
-		return TRUE;
-#ifdef ENABLE_WEBCIL
-	const char *p = strstr (assembly_name, ".webcil");
-	/* if assembly_name ends with .webcil, check if aname matches, with a .dll extension instead */
-	if (p && *(p + strlen(".webcil")) == 0) {
-		size_t n = p - assembly_name;
-		if (!strncmp (bsymfile->aname, assembly_name, n)
-			&& !strcmp (bsymfile->aname + n, ".dll"))
-			return TRUE;
-	}
-	p = strstr (assembly_name, MONO_WEBCIL_IN_WASM_EXTENSION);
-	if (p && *(p + strlen(MONO_WEBCIL_IN_WASM_EXTENSION)) == 0) {
-		size_t n = p - assembly_name;
-		if (!strncmp (bsymfile->aname, assembly_name, n)
-			&& !strcmp (bsymfile->aname + n, ".dll"))
-			return TRUE;
-	}
-#endif
-	return FALSE;
+	mono_bundled_resources_add_assembly_symbol_resource (assembly_name, raw_contents, size, NULL, NULL);
 }
 
 static MonoDebugHandle *
 open_symfile_from_bundle (MonoImage *image)
 {
-	BundledSymfile *bsymfile;
+	MonoDebugHandle *handle = NULL;
+	const uint8_t *data = NULL;
+	uint32_t size = 0;
+	if (mono_bundled_resources_get_assembly_resource_values (image->module_name, NULL, NULL, &data, &size))
+		handle = mono_debug_open_image (image, data, size);
 
-	for (bsymfile = bundled_symfiles; bsymfile; bsymfile = bsymfile->next) {
-		if (!bsymfile_match (bsymfile, image->module_name))
-			continue;
-
-		return mono_debug_open_image (image, bsymfile->raw_contents, bsymfile->size);
+#ifdef ENABLE_WEBCIL
+	size_t len = strlen (image->module_name);
+	char *extension = strrchr (image->module_name, '.');
+	/* if image's module_name ends with an acceptable extension, check if theres a bundled resource with a .dll extension instead */
+	if (extension && (!strcmp (extension, ".webcil") || !strcmp (extension, MONO_WEBCIL_IN_WASM_EXTENSION))) {
+		size_t n = extension - image->module_name;
+		char *module_name_dll_suffix = (char *)g_malloc0 (sizeof(char) * (n + 5));
+		memcpy (module_name_dll_suffix, image->module_name, len);
+		memcpy (module_name_dll_suffix + n, ".dll\0", 5);
+		if (mono_bundled_resources_get_assembly_resource_values (module_name_dll_suffix, NULL, NULL, &data, &size))
+			handle = mono_debug_open_image (image, data, size);
+
+		g_free (module_name_dll_suffix);
 	}
+#endif
 
-	return NULL;
+	return handle;
 }
 
 const mono_byte *
 mono_get_symfile_bytes_from_bundle (const char *assembly_name, int *size)
 {
-	BundledSymfile *bsymfile;
-	for (bsymfile = bundled_symfiles; bsymfile; bsymfile = bsymfile->next) {
-		if (!bsymfile_match (bsymfile, assembly_name))
-			continue;
-		*size = bsymfile->size;
-		return bsymfile->raw_contents;
+	const uint8_t *symbol_data = NULL;
+	uint32_t symbol_size = 0;
+	if (mono_bundled_resources_get_assembly_resource_values (assembly_name, NULL, NULL, &symbol_data, &symbol_size)) {
+		*size = symbol_size;
+		return (mono_byte *)symbol_data;
 	}
-	return NULL;
+
+#ifdef ENABLE_WEBCIL
```

Comment:
Is this needed? Doesn't mono_bundled_resources_get_assembly_resource_values take care of aliasing .webcil and MONO_WEBCIL_IN_WASM_EXTENSION -> .dll as part of search logic?

Reply:
Good point, removing this block